code/WorkInProgress/Ported/policetape.dm

```diff
@@ -118,36 +122,46 @@
 				var/obj/item/tape/P = new tape_type(cur)
 				P.icon_state = "[P.icon_base]_[dir]"
 			cur = get_step_towards(cur,end)
-	//is_blocked_turf(var/turf/T)
-		to_chat(usr, "<span class='notice'>You finish placing [src].</span>")
-		user.visible_message("<span class='warning'>[user] finishes placing [src].</span>") //Now you know who to whack with a stun baton
+		
+		if(start != end)//Prevent wasting charges on rolls with limited charges by just spamming use on the same tile.
+			to_chat(usr, "<span class='notice'>You finish placing [src].</span>")
+			user.visible_message("<span class='warning'>[user] finishes placing [src].</span>") //Now you know who to whack with a stun baton
+			return 1
+
+//Reset the process if dropped mid laying tape.
+/obj/item/taperoll/dropped(mob/user)
+	. = ..()
+	start = null
+	end = null
+
 
 /obj/item/taperoll/preattack(atom/target, mob/user, proximity_flag, click_parameters)
-	if(proximity_flag == 0) // not adjacent
-		return
+	..()
+	tape_door(target, user, proximity_flag)
 
-	if(istype(target, /obj/machinery/door/airlock) || istype(target, /obj/machinery/door/firedoor))
-		var/turf = get_turf(target)
 
-		if(locate(tape_type) in turf)
-			to_chat(user, "<span class='warning'>There's some tape already!</span>")
-			return 1
+/obj/item/taperoll/proc/tape_door(atom/target, mob/user, proximity_flag)
+	if(proximity_flag == 0)//Check adjacency.
+		return 0
 
+	if(istype(target, /obj/machinery/door/airlock) || istype(target, /obj/machinery/door/firedoor))	//Make sure we can tape the target.	
+		var/turf = get_turf(target)
+		
+		//Check to see if the object already has any tape of any kind on it.
+		for(var/obj/item/I in turf)
+			if(istype(I, /obj/item/tape))
+				to_chat(user, "<span class='warning'>It's already taped with [I]!</span>")
+				return 0
+		
+		//Success
 		to_chat(user, "<span class='notice'>You start placing [src].</span>")
-		if(!do_mob(user, target, 3 SECONDS))
-			return 1
-
-		if(locate(tape_type) in turf)
-			to_chat(user, "<span class='warning'>There's some tape already!</span>")
+		if(do_after(user, target, 3 SECONDS))
+			var/atom/tape = new tape_type(turf)
+			tape.icon_state = "[icon_base]_door"
+			tape.layer = ABOVE_DOOR_LAYER
+			to_chat(user, "<span class='notice'>You placed [src].</span>")
```

Comment:
we normally use the gerund form for completion messages, e.g.: "You finish placing \the [src]."

Reply:
I think this will cause a "the the". I'll test it right now.

Reply:
Nah. You're right.